### PR TITLE
Skip failing message

### PR DESCRIPTION
**Pull request by mrunge**
The generator can fail temporarily. Instead
of retrying, skip the failing message and pick it
up later.

### CODE REVIEW

File: gnocchi/incoming/redis.py [hunk 1]
```diff
@@ -184,23 +184,24 @@ def process_measures_for_sack(self, sack):
...
+            for message in p.listen():
+                if message['type'] == 'pmessage' and \
+                   message['pattern'] == pattern:
+                    # FIXME(jd) This is awful, we need a better way to extract this
+                    # Format is defined by _get_sack_name: incoming128-17
+                    yield self._make_sack(int(
+                        message['channel'].split(b"-")[-1]))
+        except ConnectionError:
+            # will try again later
+            pass
```
[the remark this answers is not in the document]
Comment (chungg) [reply gnocchi/incoming/redis.py:204]: i see. i don't have an environment to check this currently. i've actually never enable this feature before, i vaguely remember it being somewhat unstable at load but this is me grasping on stuff from 3+ years ago.

**Review by jd — DISMISSED** [review]
There are 2 commits here where it should have only one. I'm confused.

**Comment by mrunge** [comment]
Yes, thank you. I was planning to add a test here to make sure we're not missing anything here, but was distracted.

Somehow I also messed up a rebase and added https://github.com/gnocchixyz/gnocchi/pull/1164/commits/26ca0e0d6bedc5c2760293487730f072676a688c , which is already merged. I wonder how/why this got added/listed here in this PR at all.

**Comment by mrunge** [comment]
This is still WIP, and the latest push only addresses jds comment. No substantial change here so far.

**Comment by mrunge** [comment]
Closing this for now

**Pull request closed without merging**